octoprint_mrbeam/static/js/unref.js

```diff
@@ -138,7 +144,9 @@ Snap.plugin(function (Snap, Element, Paper, global) {
             ];
             for (var i = 0; i < attribute_keys.length; i++) {
                 var key = attribute_keys[i];
-                if (attr[key]) new_attrs[key] = attr[key];
+                if (elem_attr[key]) {
+                    src_attrs[key] = elem_attr[key];
+                }
```

Comment:
Shouldn't it be the other way around? 

`elem_attr[key] = src_attrs[key]`

Otherwise you would be modifying the source element instead of the newly created element.

Reply:
if you do not want the `src_attr` to replace `elem_attr` keys then you should prune `src_attrs` instead, and that can be done without using a long list of keys.

Reply:
I am not quite sure if the `src_attr` are a copy or just a pointer, which might influence the decision to change the function.

Reply:
@hungerpirat ?

Reply:
src_attr is a copy
Attributes of the placeholder element (here elem) should override attributes of the library element (here src_elem or src_attrs).Then the placeholder is replaced with a copy of the library element, this copy has all attrs of the original placeholder.

Reply:
yes but you don't need to keep a giant copy of all possible overlapping keys. Either replace the relevant keys (the keys they have in common except `id`) or remove the keys they have in common (and `id`) so that they don't overlap when updating the keys. 
I presume `uplicate.attr(src_attrs)` updates the keys? Or does it overwrite them?

Reply:
If you decide to keep the list, then you should remove `id` because it might overwrite your previous work on `src_attrs["id"]`

```js
            if (src_attrs["id"] !== undefined) {
                src_attrs["id"] = src_attrs["id"] + "_unref_" + counter;
            }
```

Reply:
Everything as planned. If the placeholder already has an ID, we keep this one.

Reply:
After all I did not want to refactor this cod here. I just wanted to fix a bug in the existing design.